invenio_app_rdm/records_ui/views/records.py

```diff
@@ -279,7 +279,27 @@ def not_found_error(error):
 
 def record_tombstone_error(error):
     """Tombstone page."""
-    return render_template("invenio_app_rdm/records/tombstone.html"), 410
+    # the RecordDeletedError will have the following properties,
+    # while the PIDDeletedError won't
+    record = getattr(error, "record", None)
+    if (record_ui := getattr(error, "result_item", None)) is not None:
+        if record is None:
+            record = record_ui._record
+
+        record_ui = UIJSONSerializer().dump_obj(record_ui.to_dict())
+
+    # render a 404 page if the tombstone isn't visible
+    if not record.tombstone.is_visible:
```

Comment:
i've removed the pretty defensive `if not record or not record.tombstone` here, as I understood the InvenioRDM philosophy is more to let unexpected cases crash.
And arriving at this point without a `record`, or with one that doesn't have a tombstone, is indeed very unexpected.

Any thoughts about that?
Ping @ntarocco @slint

Reply:
I would say that if is unexpected then it should not happen thus we will see it failing sooner ;)